File: QuickBuy.Dominio/Entities/Pedido.cs
using QuickBuy.Dominio.ObjetoDeValor;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace QuickBuy.Dominio.Entities
{
    public class Pedido : Entidade
    {
        public int Id { get; set; }
        public DateTime DataPedido { get; set; }
        public int UsuarioId { get; set; }
        public virtual Usuario Usuario { get; set; }
        public DateTime DataPevisaoEntrega { get; set; }
        public string CEP { get; set; }
        public string Estado { get; set; }
        public string Cidade { get; set; }
        public string EnderecoCompleto { get; set; }
        public int NumeroEndereco { get; set; }

        public int FormaPagamentoId { get; set; }
        public virtual FormaPagamento FormaPagamento { get; set; }

        /// <summary>
        /// Um usuario pode ter 1 ou muitos itens de pedido
        /// </summary>
        public virtual ICollection<ItemPedido> ItensPedido { get; set; }

        public override void Validate()
        {
            LimparMensagemValidacao();
            if (!ItensPedido.Any())
               AdicionarCritica("Critica - Pedido não pode ficar sem item ");

            if (string.IsNullOrEmpty(CEP))
                AdicionarCritica("Critica - CEP deve estar preenchido");

            if (FormaPagamentoId == 0)
                AdicionarCritica("Crítica - Não foi informado a forma de pagamento");
        }
    }
}

File: QuickBuy.Web/ClientApp/src/app/model/pedido.ts
import { ItemPedido } from "./itemPedido";


export class Pedido {
  //public int Id { get; set; }
  //      public DateTime DataPedido { get; set; }
  //      public int UsuarioId { get; set; }
  //      public virtual Usuario Usuario { get; set; }
  //      public DateTime DataPevisaoEntrega { get; set; }
  //      public string CEP { get; set; }
  //      public string Estado { get; set; }
  //      public string Cidade { get; set; }
  //      public string EnderecoCompleto { get; set; }
  //      public int NumeroEndereco { get; set; }

  //      public int FormaPagamentoId { get; set; }
  //      public virtual FormaPagamento FormaPagamento { get; set; }

  int: number;
  dataPedido: Date;
  usuarioId: number;
  dataPrevisaoEntrega: Date;
  cep: string;
  estado: string;
  cidade: string;
  enderecoCompleto: string;
  numeroEndereco: string;
  formaPagamentoId: number;
  itensPedido: ItemPedido[];

  constructor() {
    this.dataPedido = new Date();
    this.itensPedido = [];
  }
}
